File: Examples/Nearables/TemperatureExample/TemperatureExample/DetailViewController.m
//
//  DetailViewController.m
//  TemperatureExample
//
//  Created by Grzegorz Krukiewicz-Gacek on 24.12.2014.
//  Copyright (c) 2014 Estimote Inc. All rights reserved.
//

#import "DetailViewController.h"

@interface DetailViewController () <ESTNearableManagerDelegate>

@property (nonatomic, strong) ESTNearableManager *nearableManager;

@end

@implementation DetailViewController

- (void)viewDidLoad
{
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.

    self.nearableManager = [ESTNearableManager new];
    self.nearableManager.delegate = self;
    
    [self.nearableManager startRangingForIdentifier:self.nearable.identifier];
    
    self.typeLabel.text = [ESTNearableDefinitions nameForType:self.nearable.type];
}

#pragma mark - ESTNearableManager delegate

- (void)nearableManager:(ESTNearableManager *)manager didRangeNearable:(ESTNearable *)nearable
{
    self.temperatureLabel.text = [NSString stringWithFormat:@"%.1f°C", nearable.temperature];
}

@end

File: Examples/nearables/TemperatureExample-Swift/TemperatureExample-Swift/DetailViewController.swift
//
//  DetailViewController.swift
//  TemperatureExample-Swift
//
//  Created by Grzegorz Krukiewicz-Gacek on 24.12.2014.
//  Copyright (c) 2014 Estimote Inc. All rights reserved.
//

import UIKit

class DetailViewController: UIViewController, ESTNearableManagerDelegate {

    @IBOutlet weak var typeLabel: UILabel!
    @IBOutlet weak var temperatureLabel: UILabel!

    var nearable:ESTNearable!
    var nearableManager:ESTNearableManager!
    
    override func viewDidLoad() {
        super.viewDidLoad()

        nearableManager = ESTNearableManager()
        nearableManager.delegate = self
        nearableManager .startRangingForIdentifier(nearable.identifier)
        
        typeLabel.text = ESTNearableDefinitions.nameForType(nearable.type)
    }
    
    //MARK: - ESTNearableManager delegate
    
    func nearableManager(manager: ESTNearableManager!, didRangeNearable nearable: ESTNearable!) {
        temperatureLabel.text = NSString(format: "%.1f°C", nearable.temperature)
    }
}

